consensus/double_sign.go

```diff
@@ -70,6 +70,14 @@ func (consensus *Consensus) checkDoubleSign(recvMsg *FBFTMessage) bool {
 
 						now := big.NewInt(time.Now().UnixNano())
 
+						leaderShardKey := *shard.FromLibBLSPublicKeyUnsafe(consensus.LeaderPubKey)
```

Comment:
This FromLibBLSPublicKeyUnsafe could return nil. And deref it will crash. Better check nil first.

Reply:
yes, the Unsafe is only a convience when have invariant of never nil pointer